### PR TITLE
Added some tests for IdlGenerator

### PR DESCRIPTION
**Pull request by sigiesec**

### CODE REVIEW

File: com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/TestExtensionsTest.xtend [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+package com.btc.serviceidl.tests
+
+import org.junit.Test
+
+import static extension com.btc.serviceidl.tests.TestExtensions.*
+import static org.junit.Assert.*
+
+class TestExtensionsTest {
```
Comment (schregger) [inline com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/TestExtensionsTest.xtend]: These are the tests for "normalize". IMHO there should be test for the "assertEquals..." functions. At least, to get a high test coverage.

File: com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/generator/AbstractGeneratorTest.xtend [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import static org.junit.Assert.*
+
+class AbstractGeneratorTest {
+	@Inject extension ParseHelper<IDLSpecification>
+	@Inject IGenerator2 underTest
+	@Inject IGenerationSettingsProvider generationSettingsProvider
+
+	def void checkGenerators(CharSequence input, Set<ArtifactNature> artifactNatures, Set<ProjectType> projectTypes, int fileCount,
+		Map<String, String> contents) {
+		val spec = input.parse
```
Comment (schregger) [inline com.btc.serviceidl.tests/src/com/btc/serviceidl/tests/generator/AbstractGeneratorTest.xtend]: Where is "spec" used?

File: com.btc.serviceidl/src/com/btc/serviceidl/generator/java/JavaGenerator.xtend [hunk 3]
```diff
@@ -98,59 +101,63 @@ class JavaGenerator
...
-         generateProject(ProjectType.TEST, interface_declaration)
-         generateProject(ProjectType.SERVER_RUNNER, interface_declaration)
-         generateProject(ProjectType.CLIENT_CONSOLE, interface_declaration)
-         
-         generatePOM(interface_declaration)
+         val activeProjectTypes = Sets.intersection(projectTypes, 
+			new HashSet<ProjectType>(Arrays.asList(ProjectType.SERVICE_API, ProjectType.IMPL, ProjectType.PROTOBUF, ProjectType.PROXY,
+				ProjectType.DISPATCHER, ProjectType.TEST, ProjectType.SERVER_RUNNER, ProjectType.CLIENT_CONSOLE  
+			)))
+		 for (projectType : activeProjectTypes)
```
Comment (huttenlocher) [inline com.btc.serviceidl/src/com/btc/serviceidl/generator/java/JavaGenerator.xtend]: Maybe this loop (lines 154-157) as one single-line statement:
`activeProjectTypes.forEach[ generateProject(it, interface_declaration) ]`

File: com.btc.serviceidl/src/com/btc/serviceidl/generator/dotnet/DotNetGenerator.xtend [hunk 4]
```diff
@@ -121,33 +124,36 @@ class DotNetGenerator
...
-      generateProjectStructure(ProjectType.SERVICE_API, module)
-      generateProjectStructure(ProjectType.IMPL, module)
-      generateProjectStructure(ProjectType.PROXY, module)
-      generateProjectStructure(ProjectType.DISPATCHER, module)
-      generateProjectStructure(ProjectType.TEST, module)
+     val activeProjectTypes = Sets.intersection(projectTypes, 
+		new HashSet<ProjectType>(Arrays.asList(ProjectType.SERVICE_API, ProjectType.IMPL, ProjectType.PROXY,
+			ProjectType.DISPATCHER, ProjectType.TEST  
+		)))
+	 for (projectType : activeProjectTypes)
```
Comment (huttenlocher) [inline com.btc.serviceidl/src/com/btc/serviceidl/generator/dotnet/DotNetGenerator.xtend]: Maybe this loop (lines 153-156) as one single-line statement:
`activeProjectTypes.forEach[generateProjectStructure(it, module)]`

**Comment by sigiesec** [comment]
@schregger @huttenlocher Thanks for your suggestions. I will address them in a separate PR, after I reformatted all files.

**Pull request merged**